### PR TITLE
Voice Messaging

### LINKED ISSUE #33: Voice Messages
Sounds can be played when received.

### PR DESCRIPTION
**Pull request by CyB3RC0nN0R**
Closes #33

### CODE REVIEW

File: src/main/java/envoy/client/ui/controller/ChatScene.java [hunk 1]
```diff
@@ -171,6 +176,9 @@ public void initializeData(SceneContext sceneContext, LocalDB localDB, Client cl
 		contactLabel.setText(localDB.getUser().getName());
 		MessageControl.setUser(localDB.getUser());
 		if (!client.isOnline()) updateInfoLabel("You are offline", "infoLabel-info");
+
+		recorder = new AudioRecorder();
+		if (!recorder.isSupported()) voiceButton.setDisable(true);
```
Comment (delvh) [inline src/main/java/envoy/client/ui/controller/ChatScene.java:181]:
```suggestion
		voiceButton.setDisable(!recorder.isSupported());
```

File: src/main/java/envoy/client/data/audio/AudioRecorder.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	}
+
+	/**
+	 * Cancels the active recording.
+	 * 
+	 * @since Envoy Client v0.1-beta
+	 */
+	public void cancel() {
+		line.stop();
+		line.close();
```
Comment (DieGurke) [inline src/main/java/envoy/client/data/audio/AudioRecorder.java:117]: here should be a `Files.delete(tempFile);` inserted as well.

File: src/main/java/envoy/client/ui/AudioControl.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	 * @param audioData the audio data to play.
+	 * @since Envoy Client v0.1-beta
+	 */
+	public AudioControl(byte[] audioData) {
+		var button = new Button("Play");
+		button.setOnAction(e -> {
+			try {
+				player.play(audioData);
+			} catch (EnvoyException ex) {
+
```
Comment (DieGurke) [inline src/main/java/envoy/client/ui/AudioControl.java:35]: Insert something like an info here.

File: src/main/java/envoy/client/ui/controller/ChatScene.java [hunk 4]
```diff
@@ -230,6 +249,25 @@ private void addContactButtonClicked() {
...
+					recording = true;
+					Platform.runLater(() -> voiceButton.setText("Recording..."));
+					recorder.start();
+				} else {
+					pendingAttachment	= new Attachment(recorder.finish(), AttachmentType.VOICE);
+					recording			= false;
+					Platform.runLater(() -> { voiceButton.setText("Record Voice Message"); checkPostConditions(false); });
+				}
+			} catch (EnvoyException e) {
+				e.printStackTrace();
```
Comment (DieGurke) [inline src/main/java/envoy/client/ui/controller/ChatScene.java:266]: Rather log here

**Pull request merged**